js/viz/bar_gauge.d.ts

```diff
@@ -89,6 +89,8 @@ export interface TooltipInfo {
     target?: any;
 }
 
+/** @public */
+export type BarGaugeLabelOverlap = 'hide' | 'none' | 'shift';
```

Comment:
Do we really need introduce a new separate type for the BarGaude? Seems like this type should be used also in another components (PieChart, for example).

Reply:
Let's try to generalize it.

common/charts module contains these:

```ts
export type ChartsAxisLabelOverlap = 'rotate' | 'stagger' | 'none' | 'hide';
export type ChartsLabelOverlap = 'hide' | 'none' | 'stack';
export type LabelOverlap = 'hide' | 'none';
```

Besides that there are:
```ts
...
export type FunnelLabelOverlap = 'hide' | 'none' | 'shift';
...
export type PieChartLabelOverlap = 'hide' | 'none' | 'shift';
...
export type CircularGaugeLabelOverlap = 'first' | 'last';
```

Reply:
@vladaskorohodova, could you try to figure out how these types can be renamed so that we can move `'hide' | 'none' | 'shift'` to the common/charts module?

Reply:
Maybe `ChartsStackLabelOverlap` and `ChartsShiftLabelOverlap`? We could use something like `CommonChartsLabelOverlap` and `ExtraChartsLabelOverlap` but I think it doesn't really convey the difference. Anyway it's really hard to group Funnel, PieChart and BarGauge.

Reply:
if we allocate some separate type for Funnel, Pie Chart and Bar Gauge, we will only mislead people.
We already have a dedicated type in common/charts.d.ts Label Overlap, which is already used in several places .
Secondly, I don't see much point in combining their types into one constant due to the fact that within themselves these three components use different implementations for displaying labels

Reply:
> We already have a dedicated type in common/charts.d.ts Label Overlap
We don't have questions to `LabelOverlap` type, we just don't what maintain several different types describing the single enum.

> we allocate some separate type for Funnel, Pie Chart and Bar Gauge, we will only mislead people.
I don't understand why concrete type, for example like Vlada suggested - ChartsShiftLabelOverlap, will misslead users

>  I don't see much point in combining their types into one constant due to the fact that within themselves these three components use different implementations for displaying labels
Does not matter, there is a single implementation or several. 
Why several types for the single enum is bad:
1. users cannot save option value of the type and share it between Funnel/PieChart/BarGauge.
2. we wanna have consistent api. it's simpler to maintain it when u have only 1 type - u can change it and the change will affect all related components. U can think about this as about reduce of copypaste.

I think new common `export type ChartsShiftLabelOverlap = 'hide' | 'none' | 'shift';` is ok.

Reply:
Agreed with Vlada on the ChartsShiftLabelOverlap option